flutter_appauth/android/src/main/java/io/crossingthestreams/flutterappauth/FlutterAppauthPlugin.java

```diff
@@ -464,22 +463,42 @@ private void finishWithSuccess(Object data) {
         }
     }
 
-    private void finishWithError(String errorCode, String errorMessage, String errorDetails) {
+    private void finishWithError(String errorCode, String errorMessage, @Nullable Exception cause) {
         if (pendingOperation != null) {
-            pendingOperation.result.error(errorCode, errorMessage, errorDetails);
+            pendingOperation.result.error(errorCode, errorMessage, createErrorMap(cause));
             pendingOperation = null;
         }
     }
 
     private void finishWithDiscoveryError(AuthorizationException ex) {
-        finishWithError(DISCOVERY_ERROR_CODE, String.format(DISCOVERY_ERROR_MESSAGE_FORMAT, ex.error, ex.errorDescription), getCauseFromException(ex));
+        finishWithError(DISCOVERY_ERROR_CODE, String.format(DISCOVERY_ERROR_MESSAGE_FORMAT, ex.error, ex.errorDescription), ex);
     }
 
     private void finishWithEndSessionError(AuthorizationException ex) {
-        finishWithError(END_SESSION_ERROR_CODE, String.format(END_SESSION_ERROR_MESSAGE_FORMAT, ex.error, ex.errorDescription), getCauseFromException(ex));
+        finishWithError(END_SESSION_ERROR_CODE, String.format(END_SESSION_ERROR_MESSAGE_FORMAT, ex.error, ex.errorDescription), ex);
     }
 
-    private String getCauseFromException(Exception ex) {
+    private Map<String, Object> createErrorMap(@Nullable Exception cause) {
+        @Nullable AuthorizationException authException = cause instanceof AuthorizationException ? (AuthorizationException) cause : null;
+        Map<String, Object> responseMap = new HashMap<>();
+        responseMap.put("legacy_error_details", getCauseFromException(cause));
+
+        if (authException != null) {
+            responseMap.put("type", authException.type);
```

Comment:
An exception is being thrown as on the Dart side `type` is defined as `String?` type but the AppAuth Android SDK has `type` defined as `int` so there's a type mismatch

Reply:
`code` seems to have a similar issue and possibly other scenarios too. Can you check Android overall?

Reply:
Yeah I just noticed code as well, this does work so there must be some implicit type conversion going on somewhere which makes me uneasy. I'll go through and make sure they're being explicitly converted

Reply:
I forgot to add the link but you can see the docs for the AppAuth Android SDK when it comes to the `AuthorizationException` at https://openid.github.io/AppAuth-Android/docs/latest/net/openid/appauth/AuthorizationException.html

Reply:
> Yeah I just noticed code as well, this does work so there must be some implicit type conversion going on somewhere which makes me uneasy

I'm not seeing myself. When running the example app,  a `TypeError` being passed to the `clearBusyState()` method


<img width="1571" alt="image" src="https://github.com/user-attachments/assets/a0c9b474-23c5-488a-8500-f17fe1c8ceb7">

Reply:
That's strange, it works for me but I'm suddenly getting a type error for the user cancellation boolean, despite it all working after my initial tests... Can't think what could have changed, I'll make sure it's bulletproof anyway